src/extractor/guidance/coordinate_extractor.cpp

```diff
@@ -132,7 +132,7 @@ util::Coordinate CoordinateExtractor::ExtractRepresentativeCoordinate(
     {
         const auto result = ExtractCoordinateAtLength(
             skipping_inaccuracies_distance, coordinates);
-        BOOST_ASSERT(is_valid_result(coordinates.back()));
+        BOOST_ASSERT(is_valid_result(result));
```

Comment:
Essentially it could trigger on any dead end. Yes. Since intersection_node == to_node, it will always trigger if this case is reached.